controllers/serviceclaim_controller.go

```diff
@@ -313,6 +323,15 @@ func (r *ServiceClaimReconciler) processServiceClaim(
 	// if the number of SED keys is more than the number of secret data entries
 	// that indicates one or more keys are missing
 	if len(sclaim.Spec.ServiceEndpointDefinitionKeys) > count {
+		c := metav1.Condition{
+			LastTransitionTime: metav1.NewTime(time.Now()),
+			Type:               primazaiov1alpha1.ServiceBindingConditionReady,
+			Status:             metav1.ConditionFalse,
+			Reason:             "MissingKeyInSEDs",
+			Message:            "",
```

Comment:
I'd propose to use a less technical reason 

```suggestion
			Reason:             "NoMatchingServiceFound",
			Message:            err.Error(),
```